katana/grib_crop_wgrib2.py

```diff
@@ -1,15 +1,54 @@
 import datetime
+import glob
 import os
 from subprocess import PIPE, Popen
 
 import dateparser
 import numpy as np
 import utm
 
+fmt1 = '%Y%m%d'
+fmt2 = '%H'
 
-def grib_to_sgrib(fp_in, out_dir, file_dt, x, y, logger,
-                  buff=1500, zone_letter='N', zone_number=11,
-                  nthreads_w=1):
+
+def call_wgrib2(action, logger):
+    # run command line using Popen
+    logger.debug('Running "{}"'.format(action))
+    s = Popen(action, shell=True, stdout=PIPE,
+              stderr=PIPE, universal_newlines=True)
+
+    # stream the output of WindNinja to the logger
+    output = []
+    fatl = False
+    with s.stdout:
+        for line in iter(s.stdout.readline, ""):
+            logger.debug(line.rstrip())
+            output.append(line.rstrip())
+
+    with s.stderr:
+        # eline = s.stderr.readline()
+        for line in iter(s.stderr.readline, ""):
+            logger.warning(line.rstrip())
+            output.append(line.rstrip())
+
+            # if it failed, find a different forecast hour
+            if "FATAL" in line:
+                fatl = True
+
+    # if errors then create an exception
+    return_code = s.wait()
+    if return_code:
+        logger.warning("An error occured while running wgrib2 action")
+        # for line in output:
+        #     logger.warning(line)
+        fatl = True
+
+    return fatl
+
+
+def grib_to_small_grib(fp_in, out_dir, file_dt, x, y, logger,
+                       buff=1500, zone_letter='N', zone_number=11,
```

Comment:
May as well match the default value buff=6000 from the CoreConfig since that gets passed

                sgrib, tmp_grib, fp_out = grib_to_small_grib(
                    fp,
                    out_dir,
                    dt,
                    x1, y1,
                    logger,
                    buff=buff,
                    zone_letter=zone_letter,
                    zone_number=zone_number,
                    nthreads_w=nthreads_w)

Reply:
Changed.